content/articles/spam-detection-model-using-scikit-learn/index.md

```diff
@@ -0,0 +1,338 @@
+<!--more-->
+Spam detection helps in detecting these spam messages and comments. Spam detection models filter out unwanted messages and comments. This ensures an individual receives messages or notifications that are crucial to them.
+When building the spam detection model, we will provide the model with a dataset that consists of spam and non-spam comments.
+The model will learn from this dataset and find relevant patterns that will help it to distinguish between spam and non-spam comments. 
+
+In this tutorial, we will build a machine learning model that will detect Youtube comments as spam or non-spam. We will use a dataset that contains a list of comments from popular Youtube channels to train our model. Finally, we will implement the model using the Naive Bayes algorithm.
+
+### Table of contents
+
+- [Prerequisites](#prerequisites)
+- [Dataset preparation](#dataset-preparation)
+- [Extracting important columns](#extracting-important-columns)
+- [Feature extraction from text](#feature-extraction-from-text)
+- [Model building](#model-building)
+- [Accuracy score of our model](#accuracy-score-of-our-model)
+- [Model evaluation](#model-evaluation)
+- [Making a single prediction](#Making-a-single-prediction)
+- [Making another prediction](#making-another-prediction)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+
+A reader should know the following to have a clear understanding of this tutorial.
+
+- Be well equipped with [Python programming](/engineering-education/python-projects-for-beginners/) skills.
+- Understand the concepts of [machine learning.](/engineering-education/house-price-prediction/)
+- Have some knowledge about [natural language processing](/engineering-education/nlp-based-detection-model-using-neattext-and-scikit-learn/).
+- Know how to work with some of the [Scikit-learn](https://scikit-learn.org/stable/) algorithms.
+- Know how to build a machine learning model using [Google Colab notebook](https://research.google.com/).
+
+### Dataset preparation
+
+The dataset used contains a list of comments from popular Youtube channels. We will use a dataset collected from five Youtube channels. We need to prepare this dataset so that it can be ready for use. 
+Data preparation involves correctly formatting our dataset to make it easy for use by the model during training.
+
+First, we need to load these datasets into our machine. Let's import the packages that will load our dataset.
+
+```python
+import pandas as pd
+import numpy as np
+```
+We will use Pandas to read the datasets and Numpy will be used to perform mathematical operations on these datasets. Since we have collected the dataset from five Youtube channels, we will have a total of five datasets. 
```

Comment:
```suggestion
We will use `Pandas` to read the datasets and `Numpy`  to perform mathematical operations on these datasets. Since we have collected the dataset from five Youtube channels, we will have a total of five datasets. 
```